analysis/preprocess/preprocess_data.R

```diff
@@ -0,0 +1,157 @@
+)
+
+message("Column classes defined")
+
+# Read cohort dataset ---------------------------------------------------------- 
+
+df <- read_csv(paste0("output/input_",cohort_name,".csv.gz"), 
+               col_types = col_classes)
+
+message(paste0("Dataset has been read successfully with N = ", nrow(df), " rows"))
+
+# Format columns ---------------------------------------------------------------
+
+df <- df %>%
+  mutate(across(c(contains("_date")),
+                ~ floor_date(as.Date(., format="%Y-%m-%d"), unit = "days")),
+         across(contains('_birth_year'),
+                ~ format(as.Date(., origin = "1970-01-01"), "%Y")),
+         across(contains('_num') & !contains('date'), ~ as.numeric(.)),
+         across(contains('_cat'), ~ as.factor(.)),
+         across(contains('_bin'), ~ as.logical(.)))
+
+# Overwrite vaccination information for dummy data and vax cohort only ---------
+
+if(Sys.getenv("OPENSAFELY_BACKEND") %in% c("", "expectations") &&
+   cohort_name %in% c("vax")) {
+  source("analysis/preprocess/modify_dummy_vax_data.R")
+  message("Vaccine information overwritten successfully")
+}
+
+# Describe data ----------------------------------------------------------------
+
+sink(paste0("output/describe_",cohort_name,".txt"))
+print(Hmisc::describe(df))
+sink()
+message ("Cohort ",cohort_name, " description written successfully!")
+
+# QC for consultation variable and set max to 365 (i.e., one per day) ----------
+
+df <- df %>%
+  mutate(cov_num_consulation_rate = replace(cov_num_consulation_rate, 
+                                            cov_num_consulation_rate > 365, 365))
+
+# Define COVID-19 severity -----------------------------------------------------
+
+df <- df %>%
+  mutate(sub_cat_covid19_hospital = 
+           ifelse(!is.na(exp_date_covid19_confirmed) &
+                    !is.na(sub_date_covid19_hospital) &
+                    sub_date_covid19_hospital - exp_date_covid19_confirmed >= 0 &
+                    sub_date_covid19_hospital - exp_date_covid19_confirmed < 29, "hospitalised",
+                  ifelse(!is.na(exp_date_covid19_confirmed), "non_hospitalised", 
+                         ifelse(is.na(exp_date_covid19_confirmed), "no_infection", NA)))) %>%
+  mutate(across(sub_cat_covid19_hospital, factor))
+
+df <- df[!is.na(df$patient_id),]
+df[,c("sub_date_covid19_hospital")] <- NULL
+
+message("COVID19 severity determined successfully")
+
+# Restrict columns and save analysis dataset -----------------------------------
+
+df1 <- df %>% 
+  select(patient_id,
+         death_date,
+         starts_with("index_date_"),
+         has_follow_up_previous_6months,
+         deregistration_date,
+         starts_with("end_date_"),
+         contains("sub_"), # Subgroups
+         contains("exp_"), # Exposures
+         contains("out_"), # Outcomes
+         contains("cov_"), # Covariates
+         contains("qa_"), # Quality assurance
+         contains("step"), # diabetes steps
```

Comment:
You don't need diabetes step variables in this dataset.

Reply:
Done